packages/esm-form-entry-app/src/app/fe-wrapper/fe-wrapper.component.ts

```diff
@@ -370,4 +376,25 @@ export class FeWrapperComponent implements OnInit, OnDestroy {
       store.setState(obj);
     }
   }
+
+  /**
+   * Sets up a listener to mark the workspace as dirty when form values change.
+   *
+   * This method observes the root form control for changes. When a change occurs,
+   * it sets the 'promptBeforeClosing' flag to true, indicating that the user
+   * should be prompted before closing the workspace due to unsaved changes.
+   */
+  private setupWorkspaceDirtyStateListener(): void {
+    const promptBeforeClosing = this.singleSpaPropsService.getPropOrThrow('promptBeforeClosing');
+
+    this.form.rootNode.control.valueChanges
+      .pipe(
+        map((control) => Boolean(control)),
```

Comment:
```suggestion
        map(() => this.form.rootNode.control.dirty),
```

Maybe it'd be better to check the actual `dirty` state of the form instead of just checking for a truthy control value.